packages/api/src/controllers/generate.test.ts

```diff
@@ -0,0 +1,166 @@
+import FormData from "form-data";
+import { User } from "../schema/types";
+import {
+  AuxTestServer,
+  TestClient,
+  clearDatabase,
+  setupUsers,
+  startAuxTestServer,
+} from "../test-helpers";
+import serverPromise, { TestServer } from "../test-server";
+
+let server: TestServer;
+let mockAdminUserInput: User;
+let mockNonAdminUserInput: User;
+
+// jest.setTimeout(70000)
+
+beforeAll(async () => {
+  server = await serverPromise;
+
+  mockAdminUserInput = {
+    email: "user_admin@gmail.com",
+    password: "x".repeat(64),
+  };
+
+  mockNonAdminUserInput = {
+    email: "user_non_admin@gmail.com",
+    password: "y".repeat(64),
+  };
+});
+
+afterEach(async () => {
+  await clearDatabase(server);
+});
+
+describe("controllers/generate", () => {
+  let client: TestClient;
+  let adminUser: User;
+  let adminApiKey: string;
+  let nonAdminUser: User;
+  let nonAdminToken: string;
+
+  let aiGatewayServer: AuxTestServer;
+  let aiGatewayCalls: Record<string, number>;
+
+  beforeAll(async () => {
+    aiGatewayServer = await startAuxTestServer(30303); // port configured in test-params.ts
+    const apis = [
+      "text-to-image",
+      "image-to-image",
+      "image-to-video",
+      "upscale",
+    ];
+    for (const api of apis) {
+      aiGatewayServer.app.post(`/${api}`, (req, res) => {
+        aiGatewayCalls[api] = (aiGatewayCalls[api] || 0) + 1;
+        return res.status(200).json({
+          message: "success",
+          reqContentType: req.headers["content-type"] ?? "unknown",
+        });
+      });
+    }
+  });
+
+  afterAll(async () => {
+    await aiGatewayServer.close();
+  });
+
+  beforeEach(async () => {
+    ({ client, adminUser, adminApiKey, nonAdminUser, nonAdminToken } =
+      await setupUsers(server, mockAdminUserInput, mockNonAdminUserInput));
+
+    client.apiKey = adminApiKey;
+    await client.post("/experiment", {
+      name: "ai-generate",
+      audienceUserIds: [adminUser.id, nonAdminUser.id],
+    });
+    client.apiKey = null;
+    client.jwtAuth = nonAdminToken;
+
+    aiGatewayCalls = {};
+  });
+
+  describe("API proxies", () => {
+    it("should call the AI Gateway for generate API /text-to-image", async () => {
+      const res = await client.post("/beta/generate/text-to-image", {
```

Comment:
How expensive is this when it runs on every github actions during tests?

Reply:
We actually don't run the inference on tests! These APIs will hit the local server initialized on the `beforeAll` which is only dummy, returns a 200